### PR TITLE
DSM_mercury_fix

### PR DESCRIPTION
**Pull request by pegahtah**
fixing a bug

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/CreateClinicalDummyKitRoute.java [hunk 1]
```diff
@@ -170,8 +171,9 @@ public Object handle(Request request, Response response) {
                 logger.info("found randomOncHistoryDetailId " + randomOncHistoryDetailId);
                 logger.info("found short id " + maybeParticipantByParticipantId.get().getProfile().map(ESProfile::getHruid));
                 while (tries < 10 && (oncHistoryDetail == null || StringUtils.isBlank(oncHistoryDetail.getAccessionNumber())
-                        || maybeParticipantByParticipantId.isEmpty() || !participantIsEnrolled(maybeParticipantByParticipantId)
-                        || maybeParticipantByParticipantId.get().getProfile().map(ESProfile::getHruid).isEmpty())) {
+                        || maybeParticipantByParticipantId.isEmpty()
+                        || maybeParticipantByParticipantId.get().getProfile().map(ESProfile::getHruid).isEmpty()
+                        || !participantIsEnrolled(maybeParticipantByParticipantId))) {
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/CreateClinicalDummyKitRoute.java:178]: order of this checking should have been changed to check enrollment last (when we are sure there is a short id)

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/CreateClinicalDummyKitRoute.java [hunk 2]
```diff
@@ -95,8 +95,9 @@ public Object handle(Request request, Response response) {
             ddpParticipantId = new BSPDummyKitDao().getRandomParticipantForStudy(ddpInstance);
             maybeParticipantByParticipantId =
                     ElasticSearchUtil.getParticipantESDataByParticipantId(ddpInstance.getParticipantIndexES(), ddpParticipantId);
-            while (tries < 10 && (maybeParticipantByParticipantId.isEmpty() || !participantIsEnrolled(maybeParticipantByParticipantId)
-                    || maybeParticipantByParticipantId.get().getProfile().map(ESProfile::getHruid).isEmpty())) {
+            while (tries < 10 && (maybeParticipantByParticipantId.isEmpty()
```
Comment (devonbush) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/CreateClinicalDummyKitRoute.java:99]: can you add a comment at the top of this while to explain what you're intending to do?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/CreateClinicalDummyKitRoute.java [hunk 3]
```diff
@@ -170,8 +171,9 @@ public Object handle(Request request, Response response) {
                 logger.info("found randomOncHistoryDetailId " + randomOncHistoryDetailId);
                 logger.info("found short id " + maybeParticipantByParticipantId.get().getProfile().map(ESProfile::getHruid));
                 while (tries < 10 && (oncHistoryDetail == null || StringUtils.isBlank(oncHistoryDetail.getAccessionNumber())
```
Comment (devonbush) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/CreateClinicalDummyKitRoute.java:175]: can you add a comment at the top of this while to explain what you're intending to do?

**Review by devonbush — DISMISSED** [review]
code looks fine, comments would be appreciated :)

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/CreateClinicalDummyKitRoute.java (hunk 3 above)
Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/CreateClinicalDummyKitRoute.java:175]: Sure, this was a dummy route for GP's integrated testing, the conditions kept adding when they made requests. I agree they look a lot now. I will add comments to this code and the sequencing code.

**Review by devonbush — APPROVED** [review]

**Pull request merged**